agent/command.go

```diff
@@ -217,7 +217,7 @@ func (a *Agent) runCommandOrFunc(ctx context.Context, tc *taskContext, commandIn
 			commandSpan.SetStatus(codes.Error, "running command")
 			commandSpan.RecordError(err, trace.WithAttributes(tc.taskConfig.TaskAttributes()...))
 			commandSpan.End()
-			if commandInfo.RetryOnFailure {
+			if commandInfo.RetryOnFailure || cmd.RetryOnFailure() {
```

Comment:
Having two separate checks can get confusing without the context. To avoid needing two checks, can you please instead set commandInfo.retryOnFailure in runCommandsInBlock before passing it in to runCommandOrFunc?

Reply:
Actually [this logic](https://gist.github.com/hadjri/dac307de641cb18a9e9b605526154a72#file-registry-go-L141-L194) ensures that `cmd.RetryOnFailure()` by itself is a sufficient check since it applies to commands within a function as well as a standalone command. Changed it to only check that, and verified in staging that the command still works in both cases.